1) Добавил в ProductService новый метод isExist, реализовал его в Pro…

Kirill1504 opened the pull request:
…ductServiceImpl и добавил описание.

2) В ManagerProductsRestController, в методе, добавляющем новый товар, прописал логику для товара с таким же именем и пустым именем
3) Пофиксил баг с чекбоксом "Спрятать удаленные товары", теперь работает корректно
4) Сделал уведомление toastr для добавления товаров с незаполненными полями

Kirill-Lebedev committed:
1) Добавил в ProductService новый метод isExist, реализовал его в ProductServiceImpl и добавил описание.
2) В ManagerProductsRestController, в методе, добавляющем новый товар, прописал логику для товара с таким же именем и пустым именем
3) Пофиксил баг с чекбоксом "Спрятать удаленные товары", теперь работает корректно
4) Сделал уведомление toastr для добавления товаров с незаполненными полями

timRybin commented on src/main/java/com/jm/online_store/service/impl/ProductServiceImpl.java:
```diff
@@ -478,4 +478,18 @@ public Long editProduct(Product product) {
+     */
+    @Override
+    @Transactional
+    public boolean isExist(String productName) {
```
Можно реализовать средствами Jpa, будет попроще

timRybin commented on src/main/java/com/jm/online_store/service/impl/ProductServiceImpl.java:
```diff
@@ -478,4 +478,18 @@ public Long editProduct(Product product) {
+    @Transactional
+    public boolean isExist(String productName) {
+        Optional<Product> product = productRepository.findByProduct(productName);
+        return product.isPresent();
```
Можно сделать через репозиторий, будет попроще

Kirill1504 replied:
Не очень понял, что имеешь ввиду, я же и так сделал через репозиторий)

timRybin commented on src/main/java/com/jm/online_store/controller/rest/ManagerProductsRestController.java:
```diff
@@ -113,6 +115,15 @@ private static String getFileExtension(String myFileName) {
+            log.debug("Product with same productName already exists");
+            return new ResponseEntity("duplicatedNameProductError", HttpStatus.BAD_REQUEST);
+        }
+        if (product.getProduct().equals("")) {
```
Можно сделать проверку на фронте

Kirill1504 replied:
Да, верно, сделал проверку на пустое наименование на фронте

timRybin commented on src/main/resources/static/js/products.js:
```diff
@@ -240,6 +218,49 @@ function handleAddBtn() {
+        toastr.error('Заполните поле количества товара');
+        toastr.error('Заполните поле стоимости товара');
+        toastr.error('Заполните поле наименования товара');
+        return false;
```
Они и по отдельности нормально отработают)

Kirill1504 replied:
Не работают, если все поля пустые, выдает уведомление только о том, что нужно заполнить стоимость

timRybin commented on src/main/resources/static/js/products.js:
```diff
@@ -240,6 +218,49 @@ function handleAddBtn() {
+    if (!productToAdd.price && !productToAdd.amount) {
+        toastr.error('Заполните поле количества товара');
+        toastr.error('Заполните поле стоимости товара');
+        return false;
```
Тоже самое

Kirill1504 replied:
> После добавления/изменения/удаления продукта, чекбокс работает некорректно.
> Не знаю, насколько относиться к поставленной задаче, но при изменении/удалении продукта не всегда отображается текущая категория.
> Надо ли тестировать загрузку данных из файла?

Как я понял, такова логика. Когда меняешь удаленный товар, он восстанавливается. В чем неккоректность работы при добавлении и удалении?

timRybin reviewed (commented):
После добавления/изменения/удаления продукта, чекбокс работает некорректно.  
Не знаю насколько относиться к поставленной задаче, но при изменении/удалении продукта не всегда отображается текущая категория. 
Надо ли тестировать загрузку данных из файла?

Kirill1504 commented:
> После добавления/изменения/удаления продукта, чекбокс работает некорректно.
> Не знаю, насколько относиться к поставленной задаче, но при изменении/удалении продукта не всегда отображается текущая категория.
> Надо ли тестировать загрузку данных из файла?

Kirill1504 commented:
> После добавления/изменения/удаления продукта, чекбокс работает некорректно.
> Не знаю, насколько относиться к поставленной задаче, но при изменении/удалении продукта не всегда отображается текущая категория.
> Надо ли тестировать загрузку данных из файла?

Как я понял, такова логика, что при изменении удаленного товара, он восстанавливается. В чем некорректность работы при удалении и добавлении товар? Не очень понял

Kirill-Lebedev committed:
1) Добавил в ProductRepository и в ProductService новый метод existsProductByProduct, реализовал его в ProductServiceImpl и добавил описание.
2) Убрал в ManagerProductsRestController логику по проверке товара с пустым именем
3) Дофиксил баг с чекбоксом

Kirill-Lebedev committed:
Merge branch 'dev' into bugfix/addToastrNotificationsManagerPage

tediik commented on src/main/resources/static/js/products.js:
```diff
@@ -240,44 +218,81 @@ function handleAddBtn() {
         }
     }
 
+    /**
+     * проверяем, что наименование, цена продукта и количество != null
+     */
+    if (!productToAdd.price && !productToAdd.amount && !productToAdd.product) {
+        toastr.error('Заполните поле количества товара');
+        toastr.error('Заполните поле стоимости товара');
+        toastr.error('Заполните поле наименования товара');
+        return false;
```
этот и нижеследующий код грамотно переписать на switch
https://learn.javascript.ru/switch

Kirill-Lebedev committed:
Переписал логику вывода тостеров о заполнении полей при добавлении товара, используя switch case

tediik commented on src/main/resources/static/js/products.js:
```diff
@@ -240,44 +218,76 @@ function handleAddBtn() {
         }
     }
 
+    /**
+     * проверяем, что наименование, цена продукта и количество заполнены
+     */
+    switch (true) {
+        case (!productToAdd.price && !productToAdd.amount && !productToAdd.product) :
+            toastr.error('Заполните поле количества товара');
+            toastr.error('Заполните поле стоимости товара');
+            toastr.error('Заполните поле наименования товара');
+            return;
+
+        case (!productToAdd.price && !productToAdd.product) :
+            toastr.error('Заполните поле стоимости товара');
+            toastr.error('Заполните поле наименования товара');
+            return;
+
+        case (!productToAdd.amount && !productToAdd.product) :
+            toastr.error('Заполните поле количества товара');
+            toastr.error('Заполните поле наименования товара');
+            return;
+
+        case (!productToAdd.price && !productToAdd.amount) :
+            toastr.error('Заполните поле количества товара');
+            toastr.error('Заполните поле стоимости товара');
+            return;
+
+        case !productToAdd.price :
+            toastr.error('Заполните поле стоимости товара');
+            return;
+
+        case !productToAdd.amount :
+            toastr.error('Заполните поле количества товара');
+            return;
+
+        case !productToAdd.product :
+            toastr.error('Заполните поле наименования товара');
```
нет, не так :(

Kirill-Lebedev committed:
Поменял логику вывода тостеров

Kirill-Lebedev committed:
Добавил на бэк обратно проверку на пустое наименование продукта. Поменял логику вывода тостеров и подстветки полей при добавлении товаров

Kirill-Lebedev committed:
Добавил на бэк обратно проверку на пустое наименование продукта. Поменял логику вывода тостеров и подстветки полей при добавлении товаров(убрал закоменченный код)

Kirill-Lebedev committed:
Merge branch 'dev' into bugfix/addToastrNotificationsManagerPage

tediik committed:
Update ManagerProductsRestController.java
